AzureWebFarm.OctopusDeploy/Infrastructure/OctopusDeploy.cs

```diff
@@ -53,28 +54,52 @@ public void DeleteMachine()
         }
 
         public void DeployAllCurrentReleasesToThisMachine()
-        {
+        {            
+            const string targetRolePropertyName = "Octopus.Action.TargetRoles";
             List<TaskState> currentStatuses;
             try
             {
                 var machineId = _repository.Machines.FindByName(_machineName).Id;
                 var environment = _repository.Environments.FindByName(_config.TentacleEnvironment).Id;
+                var projects = _repository.Projects.FindAll()
+                    .Where(p => _repository.DeploymentProcesses
+                               .Get(p.DeploymentProcessId)
+                               .Steps
+                               .Any(s =>
+                                    {
+                                        string value;
+                                        return s.Properties.TryGetValue(targetRolePropertyName, out value) && value == _config.TentacleRole;
+                                    }))
+                    .Select(p => p.Id);
 
                 var dashboard = _repository.Dashboards.GetDashboard();
-                var releaseTasks = dashboard.Items
+                var dashboardItems = dashboard.Items
                     .Where(i => i.EnvironmentId == environment)
-                    .ToList()
-                    .Select(currentRelease => _repository.Deployments.Create(
-                        new DeploymentResource
-                        {
-                            Comments = "Automated startup deployment by " + _machineName,
-                            EnvironmentId = currentRelease.EnvironmentId,
-                            ReleaseId = currentRelease.ReleaseId,
-                            SpecificMachineIds = new ReferenceCollection(new[] {machineId})
-                        }
-                    ))
-                    .Select(d => d.TaskId)
+                    .Where(i => projects.Contains(i.ProjectId))
                     .ToList();
+
+                var releaseTasks = new List<string>();
+                foreach (var currentRelease in dashboardItems)
+                {
+                    try
+                    {
+                        var deployment = _repository.Deployments.Create(new DeploymentResource
+                                                       {

```

Comment:
*Pedantic*: To fit in with the coding style of this project can you de-indent this to only one indentation.